src/Sentry/SentrySdk.cs

```diff
@@ -21,20 +21,68 @@ namespace Sentry
     /// </remarks>
     public static class SentrySdk
     {
-        private static IHub _hub = DisabledHub.Instance;
+
+        /// <summary>
+        ///  Holds Hubs per thread or only mainHub if globalHubMode is disabled.
+        /// </summary>
+        private static readonly AsyncLocal<IHub> _currentHub = new AsyncLocal<IHub>();
+        /// <summary>
+        /// The Main Hub or NoOp if Sentry is disabled.
+        /// </summary>
+        private static IHub _mainHud = DisabledHub.Instance;
+
+        /// <summary>
+        /// Default value for globalHubMode is false.
+        /// </summary>
+        private const bool _globalHudDefaultMode = false;
+
+        /// <summary>
+        /// whether to use a single (global) Hub as opposed to one per thread.
+        /// </summary>
+        private static bool _globalHudMode = _globalHudDefaultMode;
 
         /// <summary>
         /// Last event id recorded in the current scope
         /// </summary>
-        public static SentryId LastEventId { [DebuggerStepThrough] get => _hub.LastEventId; }
+        public static SentryId LastEventId { [DebuggerStepThrough] get => GetCurrentHub().LastEventId; }
+
+        /// <summary>
+        /// Returns the current (threads) hub, if none, clones the mainHub and returns it.
+        /// </summary>
+        /// <returns>A Hub</returns>
+        internal static IHub GetCurrentHub()
+        {
+            if (_globalHudMode)
+            {
+                return _mainHud;
+            }
+            IHub hub = _currentHub.Value;
+            if (hub == null)
+            {
+                lock (_mainHud)
+                {
+                    hub = _mainHud.Clone();
+                    _currentHub.Value = hub;
+                }
+            }
+            return hub;
+        }
+
+        /// <summary>
+        /// Initializes the SDK while attempting to locate the DSN
+        /// </summary>
+        /// <remarks>
+        /// If the DSN is not found, the SDK will not change state.
+        /// </remarks>
+        public static IDisposable Init() => Init(DsnLocator.FindDsnStringOrDisable(), _globalHudDefaultMode);
 
         /// <summary>
         /// Initializes the SDK while attempting to locate the DSN
         /// </summary>
         /// <remarks>
         /// If the DSN is not found, the SDK will not change state.
         /// </remarks>
-        public static IDisposable Init() => Init(DsnLocator.FindDsnStringOrDisable());
+        public static IDisposable Init(bool globalHudMode = _globalHudDefaultMode) => Init(DsnLocator.FindDsnStringOrDisable(), globalHudMode);
```

Comment:
Please add an overload instead of default parameters here.

Also, annotate it with `[EditorBrowsable(EditorBrowsableState.Never)]` in an attempt to reduce adoption of direct dependency on this behavior without some wrapping package like Sentry.Xamarin